docs/blog/2023-11-20-life-of-a-sample-part-1.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Life of a sample in Thanos, and how to configure it – Ingestion – Part I
+date: "2024-09-16"
+author: Thibault Mangé (https://github.com/thibaultmg)
+---
+
+## Life of a Sample in Thanos, and How to Configure it – Ingestion – Part I
+
+### Introduction
+
+Thanos is a sophisticated distributed system with a broad range of capabilities, and with that comes a certain level of configuration complexity. In this series of articles, we will take a deep dive into the lifecycle of a sample within Thanos, tracking its journey from initial ingestion to final retrieval. Our focus will be to explain Thanos's critical internal mechanisms and highlight the essential configurations for each component, guiding you toward achieving the operational results you are aiming for. We will be covering the following Thanos components:
+
+* **Receive**: Ingests samples from Prometheus servers and prepares them for object storage.
+* **Compactor**: Merges and deduplicates blocks in object storage.
+* **Store**: Exposes blocks in object storage for querying.
+* **Query**: Retrieves data from stores and processes queries.
+* **Query Frontend**: Distributes queries to query instances.
```

Comment:
Can we add sidecar?

Reply:
Or mention that we wont cover it in this guide?

Reply:
I am covering it a bit, so I've added it here.
In the end I might also add the ruler, depending on what fits in part III. I'll add it here if needed.